src/backend/distributed/planner/multi_planner.c

```diff
@@ -75,12 +76,22 @@ CreatePhysicalPlan(Query *parse)
 	Query *parseCopy = copyObject(parse);
 	MultiPlan *physicalPlan = NULL;
 	CmdType commandType = parse->commandType;
+	bool routerPlannable = false;
 
 	if (commandType == CMD_INSERT || commandType == CMD_UPDATE ||
 		commandType == CMD_DELETE)
 	{
-		/* modifications go directly from a query to a physical plan */
-		physicalPlan = MultiModifyPlanCreate(parse);
+		routerPlannable = true;
+	}
+	else if (TaskExecutorType == MULTI_EXECUTOR_REAL_TIME)
+	{
+		routerPlannable = MultiRouterPlannableQuery(parseCopy);
```

Comment:
As @anarazel mentioned, let's make sure that `CheckNodeIsDumpable` is called on physical plan created by router planner.